### PR TITLE
[SPARK-23089][STS] Recreate session log directory if it doesn't exist

### PR DESCRIPTION
**Pull request by mgaido91**
## What changes were proposed in this pull request?

When creating a session directory, Thrift should create the parent directory (i.e. /tmp/base_session_log_dir) if it is not present. It is common that many tools delete empty directories, so the directory may be deleted. This can cause the session log to be disabled.

This was fixed in HIVE-12262: this PR brings it in Spark too.

## How was this patch tested?

manual tests

### CODE REVIEW

**Comment by jerryshao** [comment]
LGTM. So looks like the fix is exactly the same as Hive.

**Comment by jerryshao** [comment]
@cloud-fan would you please take another look on this?

**Comment by cloud-fan** [comment]
LGTM, cc @gatorsmile

**Comment by jerryshao** [comment]
@gatorsmile , do you have further comment?

File: sql/hive-thriftserver/src/main/java/org/apache/hive/service/cli/session/HiveSessionImpl.java [hunk 1]
```diff
@@ -223,6 +223,18 @@ private void configureSession(Map<String, String> sessionConfMap) throws HiveSQL
 
   @Override
   public void setOperationLogSessionDir(File operationLogRootDir) {
+    if (!operationLogRootDir.exists()) {
+      LOG.warn("The operation log root directory is removed, recreating:" +
```
Comment (jiangxb1987) [inline sql/hive-thriftserver/src/main/java/org/apache/hive/service/cli/session/HiveSessionImpl.java]: nit: extra space at the end of line

**Review by jiangxb1987 — APPROVED** [review]
LGTM

File: sql/hive-thriftserver/src/main/java/org/apache/hive/service/cli/session/HiveSessionImpl.java (hunk 1 above)
Comment (mgaido91) [reply sql/hive-thriftserver/src/main/java/org/apache/hive/service/cli/session/HiveSessionImpl.java]: sorry @jiangxb1987 I can't see any extra space at the end of this line...

Comment (mgaido91) [reply sql/hive-thriftserver/src/main/java/org/apache/hive/service/cli/session/HiveSessionImpl.java]: oh sorry, now I understand, thanks! Yes, in Hive this space is still missing, I see only now. Thanks.

**Comment by cloud-fan** [comment]
thanks, merging to master/2.3!

**Pull request closed without merging**